fairmq/shmem/UnmanagedRegion.h

```diff
@@ -263,9 +270,13 @@ struct UnmanagedRegion
 
         EventCounter* eventCounter = mngSegment.find_or_construct<EventCounter>(unique_instance)(0);
 
-        bool newShmRegionCreated = shmRegions->emplace(cfg.id.value(), RegionInfo(cfg.path.c_str(), cfg.creationFlags, cfg.userFlags, cfg.size, alloc)).second;
-        if (newShmRegionCreated) {
+        auto it = shmRegions->find(cfg.id.value());
+        if (it == shmRegions->end()) {
+            shmRegions->emplace(cfg.id.value(), RegionInfo(cfg.path.c_str(), cfg.creationFlags, cfg.userFlags, cfg.size, alloc)).second;
             (eventCounter->fCount)++;
+        } else {
+            LOG(error) << "Unamanged Region with id " << cfg.id.value() << " has already been registered. Only unique IDs per session are allowed.";
+            throw TransportError(tools::ToString("Unamanged Region with id ", cfg.id.value(), " has already been registered. Only unique IDs per session are allowed."));
         }
```

Comment:
```suggestion
        if (it != shmRegions->end()) {
            LOG(error) << "Unmanaged Region with id " << cfg.id.value() << " has already been registered. Only unique IDs per session are allowed.";
            throw TransportError(tools::ToString("Unamanged Region with id ", cfg.id.value(), " has already been registered. Only unique IDs per session are allowed."));
        }
        
        shmRegions->emplace(cfg.id.value(), RegionInfo(cfg.path.c_str(), cfg.creationFlags, cfg.userFlags, cfg.size, alloc)).second;
        (eventCounter->fCount)++;
```

Note: Even if you don't like this, there is also the "Unamanged" typo here.

Reply:
applied